### DISCUSSION TITLE
Metrics collection should default to off

### DISCUSSION THREAD

**Discussion by LeopoldTal** [4ian/GDevelop#2795] [Enhancement ideas (and their technical discussions)]
## Describe the bug
By default, metrics collection is enabled. As a result, game developers are likely to unwittingly violate data collection laws.

## To Reproduce
1. Create a simple project, not explicitly enabling or disabling metrics collection
2. Export and publish, e.g. [empty project on itch.io](https://a-fractal-teapot.itch.io/empty-gdevelop-project)
3. Open the Network tab in browser dev tools
4. Run the game
5. Wait 10 seconds
6. A request is POSTed to `https://api.gdevelop-app.com/analytics/session`

## Other details
* GDevelop 5.0.0-beta110 on OS X
* Running game in Firefox on OS X
* The action "Enable (or disable) metrics collection" works as expected.

**Comment by LeopoldTal** [comment]
Happy to do a spot of work on this, it looks [simple enough](https://github.com/4ian/GDevelop/blob/9d09ad4dd970d532eb4b4596d24a90acf0451eb3/GDJS/Runtime/runtimegame.ts#L74), but bikeshedding needed: Does this obsolete the [10-second timeout](https://github.com/4ian/GDevelop/blob/9d09ad4dd970d532eb4b4596d24a90acf0451eb3/GDJS/Runtime/runtimegame.ts#L587)? Or should metrics collection only start 10 seconds after the game is started, to ignore users who open the game but close it without really playing?

**Comment by tristanbob** [comment]
> Happy to do a spot of work on this, it looks [simple enough](https://github.com/4ian/GDevelop/blob/9d09ad4dd970d532eb4b4596d24a90acf0451eb3/GDJS/Runtime/runtimegame.ts#L74), but bikeshedding needed: Does this obsolete the [10-second timeout](https://github.com/4ian/GDevelop/blob/9d09ad4dd970d532eb4b4596d24a90acf0451eb3/GDJS/Runtime/runtimegame.ts#L587)? Or should metrics collection only start 10 seconds after the game is started, to ignore users who open the game but close it without really playing?

Based on a quick glance at your link, it does look like the default value is to NOT disable metrics.  I agree with you that GDevelop should err on the side of privacy by disabling metrics unless the user has requested them.

**Comment by 4ian** [comment]
As you I'm not a big fan of things that are invading the user privacy. On the other hand, I think analytics/metrics, when anonymous, are essential. Let me explain ;)

GDevelop has since a long time anonymous metrics reported. This is *not for tracking users*, but for knowing how much GDevelop is used:
- the overall number of users,
- what are the error messages encountered
- what are the tutorials opened the most? Same for examples?
- what are the most used export methods?
- is the asset store used? what are the assets the most used?
- what are the external editors used the most?

See how everything there is centered about understanding the usage of GDevelop.
This is not a "sexy feature", but without this, *GDevelop would be dead*: I would have stopped working on this years ago, not seeing that dozens of thousands of, often silent, users are using the app every month.
I know this might be disappointing, in an ideal world we would not need this. But it helps understanding the impact and doing the right choices.

I think the same goes for games: having these anonymous analytics built in, and home made, allow me to provide a privacy-focused way to see how a game is performing.
By anonymous, I mean: data minimization (only keep what is needed), anonymized (ip hashed for example) privacy focused (no personal identifiable information, no way to track a user across websites/games).

It's built on the same principles explained here: https://usefathom.com/data - which is GDPR compliant.
Without going into the technical details, only a very small set of information (platform name, ip, etc...) is sent to analytics endpoint. Then information are aggregated (to know the number of sessions per game). Much like the article I linked, there is also a second request that is made to measure the session time (the session itself being a not identifiable random number).
If someone was to steal all the data, they could not tell who has played anything, and even games are identified by a unique number that is random.

All of this to say: I think it's important to keep this activated, because it can help the game creator to understand the game potential, and for GDevelop I can check that we have an ever growing numbers of games having tons of sessions and be sure we're on the right track - all without storing anything private again! ;) This helps understand the impact we all have, and it's essential to continue to grow the GDevelop project.

Of course, I don't want to force things so you can just go and deactivate this (I won't disallow/prevent this), and we should probably do a better job of explaining this in the user interface.
(It's the same deal for the "splashscreen". I ask the community to keep this unless you got a subscription, which helps GDevelop to become sustainable in the future. This is just a check in the user interface - you can very well change your project by yourself and disable it. I hope people don't do that because that's not helping GDevelop).

That's a long rant already haha but I think you got the idea. 
I'm happy to make this clearer even if I made everything to make this compliant with privacy laws. I understand that it's a bit disappointing from an idealistic point of view ("it's open source it should be 100% free, sustainable by itself, without any need for any metrics" - the reality is that you need more than that to make the best game engine for everyone in the world) but I really think it's fair and beneficial for the project in the long term to keep this on by default.

**Comment by Silver-Streak** [comment]
@4ian After some reviewing of the posts on the forums from a few users over the years including some confusion around network calls/required properties, 

I think there probably should be a project-level toggle for this for transparency sake, likely in the project properties. Absolutely leave it defaulted to enabled for the reasons you list above, but provide a description of what it collects, what the benefits are, etc.
